gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/DiagnosticReportMapper.java

```diff
@@ -0,0 +1,220 @@
+        specimenCompoundsMapper.handleSpecimenChildComponents(ehrExtract, observations, observationComments, diagnosticReports);
+    }
+
+    private List<Reference> getSpecimenReferences(RCMRMT030101UK04CompoundStatement compoundStatement) {
+        return compoundStatement.getComponent()
+            .stream()
+            .map(RCMRMT030101UK04Component02::getCompoundStatement)
+            .filter(Objects::nonNull)
+            .filter(ResourceFilterUtil::isSpecimen)
+            .map(compoundStatement1 -> new Reference(new IdType(ResourceType.Specimen.name(), compoundStatement1.getId().get(0).getRoot())))
+            .toList();
+    }
+
+    private void setResultReferences(RCMRMT030101UK04CompoundStatement compoundStatement, DiagnosticReport diagnosticReport) {
+        var resultReferences = compoundStatement.getComponent()
+            .stream()
+            .map(RCMRMT030101UK04Component02::getNarrativeStatement)
+            .filter(Objects::nonNull)
+            .map(narrativeStatement -> new Reference(new IdType(ResourceType.Observation.name(), narrativeStatement.getId().getRoot())))
+            .collect(toCollection(ArrayList::new));
+
+        if (!resultReferences.isEmpty()) {
+            diagnosticReport.setResult(resultReferences);
+        }
+    }
+
+    private Optional<Observation> getObservationCommentById(List<Observation> observationComments, String id) {
+        return observationComments.stream()
+            .filter(observationComment -> id.equals(observationComment.getId()))
+            .findFirst();
+    }
+
+    private Optional<Identifier> createIdentifierExtension(List<II> id) {
+        if (id.size() > 1) {
+            final II idExtension = id.get(1);
+            if (idExtension != null && EXTENSION_IDENTIFIER_ROOT.equals(idExtension.getRoot())) {
+                return Optional.of(new Identifier()
+                    .setSystem(EXTENSION_IDENTIFIER_ROOT)
+                    .setValue(idExtension.getExtension()));
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<Reference> buildContext(List<RCMRMT030101UK04EhrComposition> compositions, List<Encounter> encounters,
+        RCMRMT030101UK04CompoundStatement compoundStatement) {
+        return Optional.ofNullable(getEncounterReference(
+            compositions,
+            encounters,
+            getCurrentEhrComposition(compositions, compoundStatement).getId().getRoot())
+        );
+    }
+
+    private Optional<InstantType> getIssued(RCMRMT030101UK04EhrExtract ehrExtract, List<RCMRMT030101UK04EhrComposition> compositions,
+        RCMRMT030101UK04CompoundStatement compoundStatement) {
+        var ehrComposition = getCurrentEhrComposition(compositions, compoundStatement);
+
+        if (authorHasValidTimeValue(ehrComposition.getAuthor())) {
+            return Optional.of(parseToInstantType(ehrComposition.getAuthor().getTime().getValue()));
```

Comment:
If author has a nullFlavor then it won't have a value probably, so this line will result in an error (from parsing null to instant type)